### PR TITLE
Issue 736/create side panel testing

### PR DESCRIPTION
**Pull request by sydrosa**
Contributes to #736 

- Adding testing to reach 100% coverage
- update `SidePanel.js` to include `hasIconOnly` for close button
- added aria label to `SidePanel` div and made `title` prop required.

### CODE REVIEW

File: ~ [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Merge remote-tracking branch 'upstream/main' into issue-736/create-side-panel-testing
```
Comment (matthewgallo) [inline ~:1]: Looks like this was committed accidentally

**Review by matthewgallo — CHANGES_REQUESTED** [review]
Now that `title` is required in `SidePanel.js`, the tests in `SidePanel.test.js` need to pass a default title so that they can pass again. See [here](https://github.com/carbon-design-system/ibm-cloud-cognitive/runs/2720382764#step:7:152)

File: ~ [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+# Please enter a commit message to explain why this merge is necessary,
+# especially if it merges an updated upstream into a topic branch.
+#
+# Lines starting with '#' will be ignored, and an empty message aborts
+# the commit.
```
Comment (davidmenendez) [inline ~:6]: what file is this?

**Review by davidmenendez — CHANGES_REQUESTED** [review]

File: ~ (hunk 2 above)
Comment (sydrosa) [reply ~:6]: Literally zero clue what this is -- will delet ⚡

Comment (matthewgallo) [reply ~:6]: Looks like VIM output after merging branches

**Review by davidmenendez — APPROVED** [review]

**Pull request merged**